examples/sparse/gcn.py

```diff
@@ -0,0 +1,92 @@
+"""
+[Semi-Supervised Classification with Graph Convolutional Networks](https://arxiv.org/abs/1609.02907)
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.data import CoraGraphDataset
+from dgl.mock_sparse import create_from_coo, diag, identity
+from torch.optim import Adam
+
+class GCN(nn.Module):
+    def __init__(self, A, in_size, out_size, hidden_size=16):
+        super().__init__()
+        # two-layer GCN
+        self.Theta1 = nn.Linear(in_size, hidden_size)
+        self.Theta2 = nn.Linear(hidden_size, out_size)
+        self.dropout = nn.Dropout(0.5)
+        # calculate the symmetrically normalized adjacency matrix
+        I = identity(A.shape, device=dev)
```

Comment:
dev is not defined?
I would suggest to calculate the norm in the main function and pass it in directly?

Reply:
change to calculate the norm in the main function